src/Microsoft.ML.ImageAnalytics/ImageLoader.cs

```diff
@@ -61,6 +61,9 @@ internal sealed class Options : TransformInputBase
 
             [Argument(ArgumentType.AtMostOnce, HelpText = "Folder where to search for images", ShortName = "folder")]
             public string ImageFolder;
+
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Whether to load images as bytes or as image type.")]
+            public bool LoadAsBytes = false;
```

Comment:
Please keep APIs consistent by creating a new entrypoint for LoadImagesAsBytes.

Reply:
Since this is for python usage, following conventions in python's scikit-learn would be more user-friendly. Following [https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_files.html](https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_files.html), the load_content option is similar to this LoadAsBytes. 

---
In reply to: [346132196](https://github.com/dotnet/machinelearning/pull/4476#discussion_r346132196) [](ancestors = 346132196)

Reply:
If I’m a C# ML NET user I will expect the same API in python as I have in C#. We have a separate API for LoadImageAsBytee in C# hence we must keep the consistency and have the same API in NimbusML.

Reply:
There has been atleast one instance where this was not followed: for ValueToKeyMappingTransformer, where the entrypoint is called Transforms.Dictionarizer. There was a good reason for creating a separate API in C#, but if there is a cleaner way of doing it for NimbusML, shouldn't we?

---
In reply to: [346455411](https://github.com/dotnet/machinelearning/pull/4476#discussion_r346455411) [](ancestors = 346455411)

Reply:
Yes, there will be at least one instance because it was a mistake. I have already given my reasoning why there should be separate API, if it does not make sense feel free to stop by and discuss offline.

---
In reply to: [346462260](https://github.com/dotnet/machinelearning/pull/4476#discussion_r346462260) [](ancestors = 346462260,346455411)

Reply:
I was the one that called it Dictionarize and it was a mistake.

---
In reply to: [346462260](https://github.com/dotnet/machinelearning/pull/4476#discussion_r346462260) [](ancestors = 346462260,346455411)

Reply:
On this change please hold on, I spoke with Gani and the answer is not clear. We will speak with Emad and get back.

---
In reply to: [346487495](https://github.com/dotnet/machinelearning/pull/4476#discussion_r346487495) [](ancestors = 346487495,346462260,346455411)

Reply:
As discussed offline with Gani and @codemzs , there will be no new entrypoint for ImageLoader as Bytes, the Entrypoint for ImageClassifier will include the image loader transform. Overall, the Image Classifier Entrypoint will take the image paths as input.